ibis/backends/pandas/tests/execution/test_timecontext.py

```diff
@@ -222,3 +263,121 @@ def test_context_adjustment_window_groupby_id(time_table, time_df3):
     # result should adjust time context accordingly
     result = expr.execute(timecontext=context)
     tm.assert_series_equal(result, expected)
+
+
+def test_construct_time_context_aware_series(time_df3):
```

Comment:
can you parameterize this? or too crazy

Reply:
I tried to parameterize this but found `expected` quite different from each other, also want to highlight differences of the existence of groupby & orderby so split these

Reply:
ok sure

if you can put a blank line fore comments (e.g. this is basically a new case). and if needed break this into separate tests to separate the cases as this is pretty long

Reply:
+1